rust-symcrypt/src/chacha.rs

```diff
@@ -88,7 +88,11 @@ mod test {
         let ct = "d31a8d34648e60db7b86afbc53ef7ec2a4aded51296e08fea9e2b5a736ee62d63dbea45e8ca9671282fafb69da92728b1a71de0a9e060b2905d6a5b67ecd3b3692ddbd7f2d778b8c9803aee328091b58fab324e4fad675945585808b4831d7bc3ff4def08e4b7a9de576d26586cec64b6116";
         let expected_tag = "1ae10b594f09e26a7e902ecbd0600691";
 
-        let (result, tag) = chacha20_poly1305_encrypt(&key, &nonce, Some(&auth_data), &pt).unwrap();
+        let nonce_array: [u8; 12] = nonce.as_slice().try_into().unwrap();
```

Comment:
Probably me not understanding Rust... why can't this be passed directly?

Reply:
`nonce` is a `Vec<u8> ` type, which is different from an array slice that forced to be 12 bytes. `&[u8; 12]`.  ( `Vec<T>` is a heap allocated and the memory is dynamically created ). Thats why we have to do `try_into()` to see if it can fit into 12 bytes, because if it wont the compiler will `panic() `and crash.